src/Models/BaseElement.php

```diff
@@ -239,7 +239,15 @@ public function onBeforeWrite()
         parent::onBeforeWrite();
 
         if (!$this->Sort) {
-            $this->Sort = static::get()->max('Sort') + 1;
+            if ($this->hasExtension(Versioned::class)) {
+                $records = Versioned::get_by_stage(BaseElement::class, Versioned::DRAFT);
+            } else {
+                $records = BaseElement::get()->max('Sort');
```

Comment:
Sorry for the late review but it appears this is a bug?

Reply:
@ScopeyNZ What seems to be the bug? I added this in as it was skipping draft version numbers and it was also using the global max rather than the local max.

Reply:
`$records = ...->max('Sort')` should return an int, but is treated like an object on the next line?

Reply:
Ah yep, but only if versioned had been removed with is rare. I'll submit a fix to the 3 branch.

Thanks

Reply:
Sweet 👍